### PR TITLE
CMake Refactor for LC Builds

### PR DESCRIPTION
**Pull request by jamesETsmith**
# General

This PR refactors the cmake build system of LAGraph so we can link to LC builds of LucataGraphBLAS. Many of these changes are intended as temporary patches to help developers build LAGraph against LucataGraphBLAS until GraphBLAS/LAGraph#161 is resolved.

# Details

The refactoring involved a couple of things:
1. Removing support for dynamic libraries (this was the main problem, I proposed a more elegant solution in GraphBLAS/LAGraph#161)
2. Turning off compilation of `experimental/benchmarks` when performing an LC build (this was due to stack frame size limits). See emusolutions/llvm-cilk#969. This is already fixed in the development versions of the toolchain and will become unnecessary after the next major toolchain release.
3. Turning off all rpath flags when performing an LC build (because our compiler doesn't support rpath flags)

# Using These Changes

My hope is that people who already have scripts to build LAGraph against LucataGraphBLAS should see no change in behavior. With that said we encourage devs to use the simplified approach made possible by this PR. See the first section of `README.md` for details. The quick version is below:

## Linking against x86 LucataGraphBLAS
```shell
cmake -B build -DGRAPHBLAS_ROOT=/path/to/LucataGraphBLAS/build/install
cmake --build build --parallel 16
```

## Linking against LC LucataGraphBLAS
If you want to compile with the Lucata toolchain (i.e. `emu-cc.sh`) then you *MUST* point cmake to an LC build of LucataGraphBLAS *AND* tell cmake to use `emu-cc.sh` as your compilers. For example:

```shell
cmake -B build_lc -DGRAPHBLAS_ROOT=/path/to/LucataGraphBLAS/build_lc/install \
    -DCMAKE_C_COMPILER=/tools/lucata/bin/emu-cc.sh \
    -DCMAKE_CXX_COMPILER=/tools/lucata/bin/emu-cc.sh
cmake --build build_lc --parallel 16
```

### CODE REVIEW

**Comment by jamesETsmith** [comment]
@moonwatcher let me know if you think this needs more documentation. I added more defaults to our build system so users should only have to specify 1-3 cmake args making build scripts unnecessary (IMHO).

File: CMakeLists.txt [hunk 1]
```diff
@@ -120,8 +120,44 @@ message ( STATUS "GraphBLAS_ROOT: ${GraphBLAS_ROOT} $ENV{GraphBLAS_ROOT}" )
...
+    set( LAGRAPH_VANILLA 1 )
+    message( STATUS "Turning on LAGRAPH_VANILLA" )
+    # Check for emu-cc in compiler path (this is fragile and should not be the long term solution)
+    string( FIND "${CMAKE_C_COMPILER}" "emu-cc" FOUND_EMU_CC )
+    if ( FOUND_EMU_CC EQUAL -1 )
+        set( LUCATA_TOOLCHAIN_FOUND FALSE )
+        message( STATUS "Compmiler IS NOT Lucata toolchain compiler emu-cc" )
+    else()
+        set( LUCATA_TOOLCHAIN_FOUND TRUE )
+        message( STATUS "Compmiler IS Lucata toolchain compiler emu-cc" )
```
Comment (mcordery) [inline CMakeLists.txt]: Spelling 'compmiler'

File: README.md [hunk 2]
```diff
@@ -3,16 +3,45 @@
...
+cmake -B build -DGRAPHBLAS_ROOT=/path/to/LucataGraphBLAS/build/install
+cmake --build build --parallel 16
+```
+
+If you want to compile with the Lucata toolchain (i.e. `emu-cc.sh`) then you *MUST* point cmake to an LC build of LucataGraphBLAS *AND* tell cmake to use `emu-cc.sh` as your compilers. For example:
+
+```shell
+cmake -B build_lc -DGRAPHBLAS_ROOT=/path/to/LucataGraphBLAS/build_lc/install \
+    -DCMAKE_C_COMPILER=/tools/lucata/bin/emu-cc.sh \
+    -DCMAKE_C_COMPILER=/tools/lucata/bin/emu-cc.sh
```
Comment (mcordery) [inline README.md]: Is this a duplicate or should one of these be DCMAKE_CXX_COMPILER?

**Review by mcordery — APPROVED** [review]
Seems ok, just one spelling thing and something that looks like a duplication

File: README.md (hunk 2 above)
Comment (jamesETsmith) [reply README.md]: Good catch, thanks!

Comment (jamesETsmith) [reply README.md]: I fixed it in the PR post too

**Pull request merged**